dialogue.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python
```

Comment:
Could consider moving this to a `src` or relevant directory. Not mandatory, since all of this is already self-contained.